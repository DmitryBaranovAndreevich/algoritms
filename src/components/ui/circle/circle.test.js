import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Circle } from "./circle";

it("Пустой компонент Circle рендерится без ошибок", () => {
  const emptyCircle = renderer.create(<Circle></Circle>).toJSON();
  expect(emptyCircle).toMatchSnapshot();
});

it("Компонент Circle с буквами рендерится без ошибок", () => {
  const textCircle = renderer.create(<Circle letter="X"></Circle>).toJSON();
  expect(textCircle).toMatchSnapshot();
});

it("Компонент Circle с text head рендерится без ошибок", () => {
  const headTextCircle = renderer.create(<Circle letter="X" head={'head'}></Circle>).toJSON();
  expect(headTextCircle).toMatchSnapshot();
});

it("Компонент Circle с React element in head рендерится без ошибок", () => {
  const headElementCircle = renderer
    .create(<Circle letter="X" head={<Circle letter="X"></Circle>}></Circle>)
    .toJSON();
  expect(headElementCircle).toMatchSnapshot();
});

it("Компонент Circle с text tail рендерится без ошибок", () => {
  const textTailCircle = renderer
    .create(<Circle letter="X" tail={"tail"}></Circle>)
    .toJSON();
  expect(textTailCircle).toMatchSnapshot();
});

it("Компонент Circle с React element in tail рендерится без ошибок", () => {
  const tailElementCircle = renderer
    .create(<Circle letter="X" tail={<Circle letter={"X"}></Circle>}></Circle>)
    .toJSON();
  expect(tailElementCircle).toMatchSnapshot();
});

it("Компонент Circle с index рендерится без ошибок", () => {
  const indexCircle = renderer
    .create(<Circle index={1}></Circle>)
    .toJSON();
  expect(indexCircle).toMatchSnapshot();
});

it("small компонент Circle  рендерится без ошибок", () => {
  const smallCircle = renderer.create(<Circle isSmall={true}></Circle>).toJSON();
  expect(smallCircle).toMatchSnapshot();
});

it("default компонент Circle  рендерится без ошибок", () => {
  const defaultCircle = renderer
    .create(<Circle state={"default"}></Circle>)
    .toJSON();
  expect(defaultCircle).toMatchSnapshot();
});

it("changing компонент Circle  рендерится без ошибок", () => {
  const changingCircle = renderer
    .create(<Circle state={"changing"}></Circle>)
    .toJSON();
  expect(changingCircle).toMatchSnapshot();
});

it("modified компонент Circle  рендерится без ошибок", () => {
  const modifiedCircle = renderer
    .create(<Circle state={"modified"}></Circle>)
    .toJSON();
  expect(modifiedCircle).toMatchSnapshot();
});