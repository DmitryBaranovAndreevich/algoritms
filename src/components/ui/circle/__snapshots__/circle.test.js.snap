// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changing компонент Circle  рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`default компонент Circle  рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`modified компонент Circle  рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`small компонент Circle  рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle с React element in head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle_text_container"
        >
          X
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    >
      X
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle с React element in tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    >
      X
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circle_text_container"
        >
          X
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Компонент Circle с index рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Компонент Circle с text head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    >
      X
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle с text tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    >
      X
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Компонент Circle с буквами рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    >
      X
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Пустой компонент Circle рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle_text_container"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
